DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c

```diff
@@ -0,0 +1,219 @@
+// Title:          Simple DMA Test
+//
+// Tool Versions:
+//
+// The design was created with the following design environment:
+// 1) Intel® Quartus® Prime 23.1 (Design includes Nios V/g which was released in
+//    23.1)
+// 2) Questasim Intel edition (Installed with Intel® Quartus® Prime)
+// 3) Intel® oneAPI DDPC++/C++ Compiler 2024.0
+// 4) Linux development host (Ubuntu* 22.04 was used, this design *should* work
+//    on Windows* but build_software.sh does not)
+//
+// Description:
+//
+// This design leverages a simple DMA oneAPI kernel that has had its RTL
+// generated and integrated into the Nios V test system that will be used to
+// control the accelerator. The DMA kernel has been configured to read from
+// memory in memory location 0 (Buffer Location 0) and write to memory in memory
+// location 1 (Buffer Location 1).
+//
+// The CPU is configured to have a single peripheral space located at
+// 0x0010_0000 and it is 1MB in size.  So if you want to connect a different
+// kernel/IP make sure to place it between data master address
+// 0x0010_0000-0x001F_FFFF.
+//
+// Author:         JCJB Date:           5/23/2023
+//
+// Version:        0.1
+//
+// Version History: 0.1:  Initial version of the design that has Nios V
+//   controlling a simple oneAPI DMA kernel that copies data from source to
+//   destination four bytes at a time.
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// This comes from the BSP and contains Nios V data cache flushing APIs
+#include <sys/alt_cache.h>
+
+// This comes from the BSP that was generated by software_build.h, it includes
+// the macros used to access the control/status registers of peripherals
+#include "io.h"
+
+// This comes from the BSP that was generated by software_build.h, it contains
+// information like address map, IRQ mapping, etc...
+#include "system.h"
+
+// including the kernel register map directly from the kernel build directory
+#include "../../../kernels/simple_dma/build/simple_dma.report.prj/register_map_offsets.hpp"
+
+// In bytes, must be a multiple of 4.  Keep it a small number to shorten the
+// simulation time and do not exceed the 1MB memory size (remember this code is
+// in there too). Once the DMA gets going this buffer will fly by fast so
+// setting it too high mostly affects the memory initialization and the
+// correctness check at the end once the DMA gets going this buffer will fly by
+// fast so setting it too high mostly affects the memory initialization and the
+// correctness check at the end.
+#define BUFFER_LENGTH 1024
+
+// Error numbers
+#define TEST_PASS 0
+#define TEST_FAIL 1
+
+/// NIOS V limits the offset of the `IOWR32...` macros to 12 bits, so explicitly
```

Comment:
This comment block is still true, but I think you want to state something like "calculating DMA kernel register offsets from system.h and register_map_offsets.hpp" since the reason why you added this was to give the DMA registers nicer names.  Then when the IOWR/IORD macros are invoked could could comment "using offsets of 0 because the kernel control and status register offsets have already been added" (or just change the IOWR and IORD to pointer deferences instead).